### PR TITLE
Added 'workgroup' as optional argument to athena component 

### PR DESCRIPTION
**Pull request by LeonardAukea**
@Jeffwan Not sure if you want this here. But at least we needed this option. If you want me to commit more changes etc. I can do that too. I noticed that you had some todos.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/3254)

### CODE REVIEW

**Comment by Jeffwan** [comment]
Thanks for the contribution!

File: components/aws/athena/query/src/query.py [hunk 1]
```diff
@@ -24,16 +24,29 @@ def get_client(region=None):
...
-      QueryExecutionContext={
-          'Database': database
-          },
-      ResultConfiguration={
-          'OutputLocation': output,
-          }
-      )
+def query(client, query, database, output, workgroup=None):
+
+  if workgroup:
```
Comment (Jeffwan) [inline components/aws/athena/query/src/query.py]: Is there more elegant way to handle optional arguments here? If we have extra configs, we will have lots of if else and nested if else logic here.

Comment (LeonardAukea) [reply components/aws/athena/query/src/query.py]: Yeah, sorry about that. The camelcase args aree not super elegant either (boto3). 

I can change it to: 

```python
arg_dict = dict(QueryString=query,
        QueryExecutionContext={
            'Database': database
            },
        ResultConfiguration={
            'OutputLocation': output,
            }
        )
    if workgroup:
        arg_dict.update(dict(WorkGroup=workgroup))

    response = client.start_query_execution(**arg_dict)
``` 

Unless you have a better option?

Comment (LeonardAukea) [reply components/aws/athena/query/src/query.py]: I accidentally added black formatting in the latest commit. Sorry in advance. if the changes are not as clear. But it's essentially the above. Except `arg_dict` is renamed to `params`

**Comment by LeonardAukea** [comment]
@Jeffwan  If more AWS components are of interest to you guys. We could potentially add more. eg. S3 interactions. Since, we are writing them anyways. It's not too much of a hassle to make them a little neater and general enough. Are you planning to provide a set of "official" AWS reusable components?

File: components/aws/athena/query/src/query.py [hunk 2]
```diff
@@ -20,73 +20,101 @@
...
+
+
+def query(client, query, database, output, workgroup=None):
+    """Executes an AWS Athena query."""
+    params = dict(
+        QueryString=query,
+        QueryExecutionContext={"Database": database},
+        ResultConfiguration={"OutputLocation": output,},
+    )
+    if workgroup:
```
Comment (Jeffwan) [inline components/aws/athena/query/src/query.py]: Thanks. This looks better.

**Comment by Jeffwan** [comment]
> @Jeffwan If more AWS components are of interest to you guys. We could potentially add more. eg. S3 interactions. Since, we are writing them anyways. It's not too much of a hassle to make them a little neater and general enough. Are you planning to provide a set of "official" AWS reusable components?

That would be great. Could you create an issue and share the plans of components you like to add? Yes, managed services components are something we'd like to add and maintain. Current focus is more on the data plane and ML area.

**Pull request merged**